ddtrace/tracer/option.go

```diff
@@ -138,6 +138,24 @@ type config struct {
 
 	// enabled reports whether tracing is enabled.
 	enabled bool
+
+	// postProcessor holds the function used to process finished spans of a trace.
+	// It reports whether the trace should be dropped.
+	postProcessor func([]ddtrace.ReadWriteSpan) bool
+}
+
+// WithPostProcessor enables processing finished spans of a trace by f. Spans
+// can be read from and written to in the processor by using the methods specified by
+// the ReadWriteSpan interface. f reports whether the trace should be dropped (dropped
```

Comment:
I may be misunderstanding what you meant here 😅, but I think that saying:

> by using methods of the `Span` within the `ReadWriteSpan`

could lead to confusion and users will try to do this:
```
func processor(spans []ddtrace.ReadWriteSpan) bool {
	for _, span := range spans {
		span.(ddtrace.Span).SetTag("key", "val")
	}
	return true
}
```
rather than:
```
func processor(spans []ddtrace.ReadWriteSpan) bool {
	for _, span := range spans {
		span.SetTag("key", "val")
	}
	return true
}
```

What do you think about this wording:
```
WithPostProcessor allows certain span fields to change after a trace has finished, and can force a trace to be dropped despite any existing sampling rules.
Modifying the span can be done by using methods of the `ReadWriteSpan` interface, such as `SetTag`.
If f returns false, the entire trace will be dropped. If f returns true, then no changes to the sampling method will be made based on the post processor function, though the trace may be dropped in the agent due to other sampling rules.
Note that dropping the trace may result in inaccurate stats if client-side stats is not enabled, since the trace will never reach the Datadog Agent, and that modifying any of the ReadWriteSpans outside the scope of the processor
function can and will likely cause a panic.
```
?

Reply:
That looks great

Reply:
While making the change, thought of a couple nits:

- To be more precise, I think we should say `WithPostProcessor sets post processor function f, which allows...` rather than `WithPostProcessor allows...`.
- Seing that client side stats is still behind a feature flag, and how to enable this FF (by setting `discovery`) is not documented anywhere, perhaps we should hold off on saying `if client-side stats is not enabled` and stick with `dropped traces and spans will not be accounted for in APM stats` ? We will then switch the wording here once client side stats are GA/ documented ?

Reply:
cc @katiehockman let me know if you agree with the two points above ? 🙂